### DISCUSSION TITLE
show a specific value among options when no result is found

### DISCUSSION THREAD

**Discussion by sobition** [JedWatson/react-select#5295] [Q&A]
Hi everyone

I am wondering if there is a solution to return a default value which is one of my options passed to the Select component when nothing matches the user input without writing the search logic from scratch. For example I have 
`[
  {label:"apple", value:"apple"},
  {label:"banana", value:"banana"},
  {label:"grape", value:"grape"},
  {label:"melon", value:"melon"},
]`
but if the user types in something that does not match any of the above I want to show the apple

**Comment by manjushsh** [comment]
@SobhanAminnejad You can achieve this using `noOptionsMessage` prop. So basically you can set option and return Custom message.
[Code](https://github.com/JedWatson/react-select/blob/4b8468636bcfadf0cfe45f9a7a6c1db5dca08d9a/packages/react-select/src/Select.tsx#L213)
[Example. Refer Last Select](https://codesandbox.io/s/react-select-multiple-old-forked-6picqk?file=/src/reffed-select.tsx:777-812)

**Reply by sobition** [reply]
Actually with your response it jumped to my mind to return a <component.Option/> as the message. it looks what I like to be but the only problem is that it's not behaving as an option. Which means by clicking it it won't be selected. I tried to define an onClick method for it too but it's not working. Here is what I tried. Would be thankful if you could help me further


<img width="585" alt="Screenshot 2022-08-02 at 13 42 10" src="https://user-images.githubusercontent.com/33763156/182366263-02a027cc-aaf8-468c-95e9-49741d2e6635.png">

**Reply by manjushsh** [reply]
@SobhanAminnejad Not a good solution but you can simulate using ref.
[Updated Code. Last Select](https://codesandbox.io/s/react-select-multiple-old-forked-6picqk?file=/src/reffed-select.tsx)
